cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -692,6 +694,17 @@ private SubmitJobRequest getSubmitJobRequest(RuntimeJobInfo runtimeJobInfo,
     GetClusterRequest getClusterRequest = GetClusterRequest.newBuilder()
         .setClusterName(clusterName).setProjectId(projectId).setRegion(region).build();
 
+    int driverCores = Integer.parseInt(provisionerProperties.getOrDefault(
+        DataprocUtils.DRIVER_VCORES, DataprocUtils.DRIVER_VCORES_DEFAULT));
+    int driverMemoryMb = Integer.parseInt(provisionerProperties.getOrDefault(
+        DataprocUtils.DRIVER_MEMORY_MB, DataprocUtils.DRIVER_MEMORY_MB_DEFAULT));
+    // client mode means the Workflow driver/Spark client will run within the RuntimeJob,
+    // so it needs at least the memory/vcpu specified for the program.
+    if (launchMode == LaunchMode.CLIENT) {
+      driverCores = Math.max(runtimeJobInfo.getVirtualCores(), driverCores);
+      driverMemoryMb = Math.max(runtimeJobInfo.getMemoryMb(), driverMemoryMb);
```

Comment:
This config will only be used when driver node pools is enabled on the dataproc cluster.

Do you think it makes sense to check if the master node has enough memory and cores required to run the job before submitting the job when driver pools is disabled? This will fail the pipeline sooner.

Reply:
there isn't actually a way to set the memory for the DataprocJob (at least I couldn't find any) without driver node pools, so there isn't really anything to check that's different than when a normal job is submitted. The workflow driver shouldn't be using much memory unless there's an action plugin doing silly stuff like loading a bunch of data into memory.

Was thinking about only allowing this when driver pools are enabled, but driver pools don't really make sense for ephemeral clusters, so it seemed a bit too restrictive since it appears to not have memory problems in most situations.